tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/hyperlink/HyperlinkPluginTest.java

```diff
@@ -0,0 +1,128 @@
+
+	@Inject
+	private ShippedCodeInitializeTestHelper shippedCodeInitializeTestHelper;
+
+	@Inject
+	private ExternalLibrariesActionsHelper externals;
+
+	@Inject
+	private StatusHelper statusHelper;
+
+	@Inject
+	private N4JSHyperlinkDetector hyperlinkDetector;
+
+	@Inject
+	private IURIEditorOpener uriEditorOpener;
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+		shippedCodeInitializeTestHelper.setupBuiltIns();
+	}
+
+	@After
+	@Override
+	public void tearDown() throws Exception {
+		// clear library manager to avoid confusing tests being executed after this test class
+		final MultiStatus multistatus = statusHelper
+				.createMultiStatus("Status of deleting NPM packages from library manager.");
+		externals.maintenanceDeleteNpms(multistatus);
+
+		shippedCodeInitializeTestHelper.tearDownBuiltIns();
+		super.tearDown();
+	}
+
+	/**
+	 * The test invokes the method
+	 * {@link N4JSHyperlinkDetector#detectHyperlinks(org.eclipse.jface.text.ITextViewer, IRegion, boolean)} to gather
+	 * all hyperlinks on the call to 'process' of "n4js-runtime-node" and checks that the returned hyperlink uri is a
+	 * file based uri.
+	 */
+	@Test
+	public void testHyperlinks() throws CoreException {
+		File prjDir = new File(getResourceUri(PROBANDS, SUBFOLDER));
+		IProject project = ProjectTestsUtils.importProject(prjDir, PROJECT_NAME);
+		IResource resourceABC = project.findMember("src/ABC.n4js");
+		IFile fileABC = ResourcesPlugin.getWorkspace().getRoot().getFile(resourceABC.getFullPath());
+		IResourcesSetupUtil.fullBuild();
+		waitForAutoBuild();
+
+		IWorkbenchPage page = EclipseUIUtils.getActivePage();
+		XtextEditor editorABC = openAndGetXtextEditor(fileABC, page);
+		ISourceViewer sourceViewer = editorABC.getInternalSourceViewer();
+		IRegion region = new Region(367, 0);
+		IHyperlink[] hlinksInABC = hyperlinkDetector.detectHyperlinks(sourceViewer, region, true);
+
+		assertTrue("Hyperlink missing", hlinksInABC != null && hlinksInABC.length == 1);
+		assertTrue("Hyperlink must be of type XtextHyperlink", hlinksInABC[0] instanceof XtextHyperlink);
+		XtextHyperlink hyperlinkToProcess = (XtextHyperlink) hlinksInABC[0];
+		URI uriProcess = hyperlinkToProcess.getURI();
+		assertTrue("Hyperlink URI must be a file uri", uriProcess.isFile());
+
+		XtextEditor editorProcess = (XtextEditor) uriEditorOpener.open(uriProcess, true);
+		sourceViewer = editorProcess.getInternalSourceViewer();
+		region = new Region(556, 12);
+		IHyperlink[] hlinksInProcess = hyperlinkDetector.detectHyperlinks(sourceViewer, region, true);
+ 
+		assertTrue("Hyperlink in external library missing", hlinksInProcess != null && hlinksInProcess.length == 1);
+		assertTrue("Hyperlink must be of type XtextHyperlink", hlinksInProcess[0] instanceof XtextHyperlink);
+		XtextHyperlink hyperlinkToEvent = (XtextHyperlink) hlinksInProcess[0];
+		URI uriEvent = hyperlinkToEvent.getURI();
+		assertTrue("Hyperlink URI must be a file uri", uriEvent.isFile());
```

Comment:
in addition to this assertion, we could test that URI `uriEvent` can actually be opened and the correct region is selected in the newly opened editor (by using the same infrastructure methods that the original hyperlink code is using; for testing the selection there is a public method`#getSelection()` in the editor, I think).

Reply:
👍  I added further logic to assert that the selection text is as expected.